Compile Errors on macOS 10.15 (Xcode 11.3)

Thank you for the library! Very interesting and would like to try it out, but running into issues compiling the code and unit tests.

Using macOS 10.15 with Xcode 11.3, trying to compile the code with unit tests results in errors:

```
[ 20%] Building CXX object CMakeFiles/UnitTest.dir/test/unit/mserialize/roundtrip.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_TEST_NO_MAIN -DBOOST_UNIT_TEST_FRAMEWORK_DYN_LINK -I/tmp/build/binlog/bin -I/tmp/build/binlog/include -isystem /usr/local/include  -O2 -g -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk   -Wall -Wextra -Werror -pedantic -Wconversion -Wsign-conversion -Wold-style-cast -Wshadow -std=c++14 -o CMakeFiles/UnitTest.dir/test/unit/mserialize/roundtrip.cpp.o -c /tmp/build/binlog/test/unit/mserialize/roundtrip.cpp
In file included from /tmp/build/binlog/test/unit/mserialize/roundtrip.cpp:6:
In file included from /tmp/build/binlog/include/mserialize/serialize.hpp:4:
/tmp/build/binlog/include/mserialize/detail/Serializer.hpp:39:5: error: static_assert failed due to requirement
      'always_false<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > > >::value' "T is not serializable"
    static_assert(always_false<T>::value, "T is not serializable");
    ^             ~~~~~~~~~~~~~~~~~~~~~~
/tmp/build/binlog/include/mserialize/serialize.hpp:34:32: note: in instantiation of function template specialization
      'mserialize::detail::BuiltinSerializer<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >, void>::serialize<OutputStream>'
      requested here
  detail::Serializer<T>::type::serialize(in, ostream);
                               ^
/tmp/build/binlog/include/mserialize/detail/Serializer.hpp:133:19: note: in instantiation of function template specialization
      'mserialize::serialize<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >, OutputStream>' requested here
      mserialize::serialize(elem, ostream);
                  ^
/tmp/build/binlog/include/mserialize/detail/Serializer.hpp:109:5: note: in instantiation of function template specialization
      'mserialize::detail::BuiltinSerializer<std::__1::vector<bool, std::__1::allocator<bool> >, void>::serialize_elems<OutputStream>' requested here
    serialize_elems(is_sequence_batch_serializable<const Sequence>{}, s, size32, ostream);
    ^
/tmp/build/binlog/include/mserialize/serialize.hpp:34:32: note: in instantiation of function template specialization
      'mserialize::detail::BuiltinSerializer<std::__1::vector<bool, std::__1::allocator<bool> >, void>::serialize<OutputStream>' requested here
  detail::Serializer<T>::type::serialize(in, ostream);
                               ^
/tmp/build/binlog/test/unit/mserialize/roundtrip.cpp:45:15: note: in instantiation of function template specialization 'mserialize::serialize<std::__1::vector<bool,
      std::__1::allocator<bool> >, OutputStream>' requested here
  mserialize::serialize(in, ostream);
              ^
/tmp/build/binlog/test/unit/mserialize/roundtrip.cpp:279:3: note: in instantiation of function template specialization '(anonymous
      namespace)::roundtrip_into<std::__1::vector<bool, std::__1::allocator<bool> >, std::__1::vector<bool, std::__1::allocator<bool> > >' requested here
  roundtrip_into(in, out);
  ^
In file included from /tmp/build/binlog/test/unit/mserialize/roundtrip.cpp:6:
In file included from /tmp/build/binlog/include/mserialize/serialize.hpp:4:
/tmp/build/binlog/include/mserialize/detail/Serializer.hpp:44:5: error: static_assert failed due to requirement
      'always_false<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > > >::value' "T is not serializable"
    static_assert(always_false<T>::value, "T is not serializable");
    ^             ~~~~~~~~~~~~~~~~~~~~~~
/tmp/build/binlog/include/mserialize/serialize.hpp:46:39: note: in instantiation of member function
      'mserialize::detail::BuiltinSerializer<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >, void>::serialized_size' requested here
  return detail::Serializer<T>::type::serialized_size(in);
                                      ^
/tmp/build/binlog/include/mserialize/detail/Serializer.hpp:156:29: note: in instantiation of function template specialization
      'mserialize::serialized_size<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > > >' requested here
      result += mserialize::serialized_size(elem);
                            ^
/tmp/build/binlog/include/mserialize/detail/Serializer.hpp:121:12: note: in instantiation of member function 'mserialize::detail::BuiltinSerializer<std::__1::vector<bool,
      std::__1::allocator<bool> >, void>::sizeof_elems' requested here
         + sizeof_elems(std::is_arithmetic<sequence_data_t<const Sequence>>{}, s);
           ^
/tmp/build/binlog/include/mserialize/serialize.hpp:46:39: note: in instantiation of member function 'mserialize::detail::BuiltinSerializer<std::__1::vector<bool,
      std::__1::allocator<bool> >, void>::serialized_size' requested here
  return detail::Serializer<T>::type::serialized_size(in);
                                      ^
/tmp/build/binlog/test/unit/mserialize/roundtrip.cpp:48:57: note: in instantiation of function template specialization 'mserialize::serialized_size<std::__1::vector<bool,
      std::__1::allocator<bool> > >' requested here
  BOOST_TEST(std::size_t(stream.tellp()) == mserialize::serialized_size(in));
                                                        ^
/tmp/build/binlog/test/unit/mserialize/roundtrip.cpp:279:3: note: in instantiation of function template specialization '(anonymous
      namespace)::roundtrip_into<std::__1::vector<bool, std::__1::allocator<bool> >, std::__1::vector<bool, std::__1::allocator<bool> > >' requested here
  roundtrip_into(in, out);
  ^
2 errors generated.
make[2]: *** [CMakeFiles/UnitTest.dir/test/unit/mserialize/roundtrip.cpp.o] Error 1
make[1]: *** [CMakeFiles/UnitTest.dir/all] Error 2
make: *** [all] Error 2
```

Thanks!

--- a/.github/workflows/on_pull_request.yml
+++ b/.github/workflows/on_pull_request.yml
@@ -90,19 +90,39 @@ jobs:
           cd Clang
           ctest -VV
 
-  # do not test on macOS: boost is not installed,
-  # installing it from brew takes too long
   build_mac_clang:
     name: Build on macOS with Clang
     runs-on: macos-latest
     steps:
       - uses: actions/checkout@v2
+      - name: Cache Boost
+        id: cache-boost
+        uses: actions/cache@v1
+        with:
+          path: boost_1_72_0/installdir
+          key: osx-boost_1_72_0
+      - name: Install Boost
+        if: steps.cache-boost.outputs.cache-hit != 'true'
+        run: |
+          wget https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.gz -Oboost.tar.gz
+          tar -zxf boost.tar.gz
+          cd boost_1_72_0
+          mkdir -p installdir
+          ./bootstrap.sh --prefix=./installdir --with-libraries=filesystem,system,test
+          ./b2 install
       - name: Build
         run: |
+          cd boost_1_72_0/installdir
+          BOOST_ROOT=$(pwd)
+          cd ../../
           mkdir Release
           cd Release
-          cmake ..
+          cmake -DBOOST_ROOT=${BOOST_ROOT} -DBoost_ARCHITECTURE=-x64 -DBINLOG_FORCE_TESTS=ON ..
           make -j2 VERBOSE=1
+      - name: Test
+        run: |
+          cd Release
+          ctest -VV
 
   build_win_msvc:
     name: Build on Windows with MSVC

--- a/include/mserialize/detail/Serializer.hpp
+++ b/include/mserialize/detail/Serializer.hpp
@@ -122,6 +122,8 @@ struct BuiltinSerializer<Sequence, enable_spec_if<
   }
 
 private:
+  using value_type = sequence_value_t<Sequence>;
+
   template <typename OutputStream>
   static void serialize_elems(
     std::false_type /* no batch copy */,
@@ -130,7 +132,7 @@ private:
   {
     for (auto&& elem : s)
     {
-      mserialize::serialize(elem, ostream);
+      mserialize::serialize<value_type>(elem, ostream);
     }
   }
 
@@ -153,7 +155,7 @@ private:
     std::size_t result = 0;
     for (auto&& elem : s)
     {
-      result += mserialize::serialized_size(elem);
+      result += mserialize::serialized_size<value_type>(elem);
     }
     return result;
   }

